### DISCUSSION TITLE
[Question][Hw5] Shape of branch predicate

### DISCUSSION THREAD

**Discussion by jirheee** [prosyslab-classroom/cs348-information-security#596] [Q&A]
Can we assume that variable comes at first operand of predicate like x < 10? Or should we consider something like 2<x?

**Comment by RiceBiscuits** [comment]
Please assume that the both cases are possible.

**Comment by jirheee** [comment]
Oh then should we consider something like 2>6 in the predicate?

**Comment by seyoungsong** [comment]
Should we consider something like (x > y) in the predicate? I think it is not possible ...

**Comment by RiceBiscuits** [comment]
@jirheee @seyoungsong Regardless of cases, you should just concentrate which an abstract type is assigned for the right or the left side of the predicate.

**Comment by jirheee** [comment]
Thank you!

**Comment by jirheee** [comment]
How about cases like x+y>5? This seem to be a lot harder....

**Comment by RiceBiscuits** [comment]
Please see corresponding *.ll file. Your analyzer reads *.ll file instead of *.c file. 
IR representation of `x+y>5` will be consisted of separate code lines.
For example, %add = add x y, and %cmp = cmp sgt %add 5.

**Comment by jirheee** [comment]
Oh now I got it Thank you!
